### ISSUE TITLE
callback results 

### ISSUE THREAD

**Issue by iplayfast** [joaomdmoura/crewAI#274]
I was following along a video tutorial https://www.youtube.com/watch?v=Jl6BuoXcZPE&ab_channel=codewithbrandon 
and everything was working fine. I subsituted ollama for openai but things seemed to be working except for the very last step. 
The callback function for the last step is missing the task_output.result.

How do I debug this? Where is the callback documentation. 
Full source at https://github.com/iplayfast/OllamaPlayground/tree/main/crewAIAsync

```
def save_markdown(task_output):
    #get today's date in the format YYY-MM-DD
    today_date = datetime.now().strftime('%Y-%m-%d')
    # set the filename with today's date
    filename = f"{today_date}.md"
    #write the task output to the markdown file
    with open(filename,'w') as file:
        file.write(task_output.result)########## Error here
    print(f"Newsletter saved as {filename}")
```

**Comment by joaomdmoura** [comment]
what version of crewai are you using?
because you might want to do `task_output.raw_output`
or you can even have the task save the output by setting a
`output_file=filename.md` to the task
--
João Moura
@joaomdmoura <http://twitter.com/joaomdmoura>


Em seg., 26 de fev. de 2024 às 04:39, Chris Bruner ***@***.***>
escreveu:

**Comment by iplayfast** [comment]
Well that was interesting ```
I'm unable to create a comprehensive AI newsletter for you as the required tool "NewsletterCompiler" is not available. However, I can suggest you delegate this 
task to a team member who has expertise in creating newsletters, and provide them with the necessary context and examples for reference. Here's an example messa
ge for delegating the work:

```
Subject: Request for creating a comprehensive AI newsletter

Hi [Team Member],

I hope this message finds you well. I have been tasked with creating a weekly AI newsletter for our internal team and external subscribers. The newsletter shoul
d include accurate, well-formatted summaries for each AI news story from the provided list. Each summary should provide sufficient context for our readers and m
aintain a consistent style and layout throughout the newsletter as shown in the example output.

Could you please create a comprehensive AI newsletter in markdown format based on this description and the following examples? I believe your expertise in creat
ing newsletters would be greatly beneficial to this project.

If you have any questions or need additional context, feel free to ask. I will ensure that all necessary information is provided.

Looking forward to your contribution!
```

Please modify the message as per your specific needs and team communication style. Let me know if you require further assistance.
```

**Comment by lvenkataraman** [comment]
Quick question.  If I am using different agents to accomplish different custom task executions how to make the Crew route to appropriate agent and get the answer and complete the execution.  I tried to use Process.hierarchical thinking if I have multiple agents connected to the process then it will execute the agent and once it gets the answer it will come back.  But it takes the output and calls the other agent which is the not the right behaviour in my case because it will get wrong answers.

If question a call Agent A; workflow done
If question b call Agent B; workflow done
if question c call Agent C; workflow done

In other words I need the Process to operate like a smart router based on the question.  Is it possible in CrewAI?

**Comment by MorphSeur** [comment]
Hello @lvenkataraman,

Any additional info about your issue?
I am facing the same error.

Thanks in advance!

**Comment by RicardoFRocha** [comment]
any news?
